paimon-filesystems/paimon-cosn/src/main/java/org/apache/paimon/cosn/COSNLoader.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.cosn;
+
+import org.apache.paimon.catalog.CatalogContext;
+import org.apache.paimon.fs.FileIO;
+import org.apache.paimon.fs.FileIOLoader;
+import org.apache.paimon.fs.Path;
+import org.apache.paimon.fs.PluginFileIO;
+import org.apache.paimon.plugin.PluginLoader;
+
+/** A {@link PluginLoader} to load cosn. */
+public class COSNLoader implements FileIOLoader {
+
+    private static final String COSN_JAR = "paimon-plugin-cosn.jar";
+
+    private static final String COSN_CLASS = "org.apache.paimon.cosn.COSNFileIO";
+
+    // Singleton lazy initialization
+
+    private static PluginLoader loader;
+
+    private static synchronized PluginLoader getLoader() {
+        if (loader == null) {
+            // Avoid NoClassDefFoundError without cause by exception
+            loader = new PluginLoader(COSN_JAR);
```

Comment:
Why is `COSN_JAR` fixed? Is it necessary to make the name to the final `paimon-plugin-cosn.jar`? Other jar name could not?

Reply:
Here I refer to the implementation of OSS, mainly to the code style can be unified